### PR TITLE
STR-19 | BE: Create UserRoles Table

### PR DESCRIPTION
**Pull request by frlzjosh**
# Description
'Write a meaningful description of your task'
   Create UserRoles Table through the service contracts model.
   Implemented functions in my Controller through my UserRoleContract interface.
   Create tests to test my logic and what I am returning
# Testing
'What does the reviewer need to do to test your PR?'
   Type this into your url `http://localhost:8080/get_role/nr_cyndee.bellamy@csun.edu` to see the response that get sent back to you.
   #What to expect:
 ```{
  "user_role": "comparison",
  "user_email": "nr_cyndee.bellamy@csun.edu",
  "display_name": "Cyndee Bellamy",
  "password": "yungbasedgod",
  "remember_token": "12394iresjntgruirf9eiokwj"
}
```

# Installation
'Are there any dependencies the reviewer needs to download?'
   You will not need to install anything. However you will need to run `phpunit` to make sure my test passes.

### CODE REVIEW

**Review by matthewfritz — DISMISSED** [review]
Please do not use a local table for this. Use nemo.memberships instead for leveraging application roles. The parent_entities_id value should be applications:istart. Remember that this is only for APPLICATION roles. If you need to figure out whether or not someone is faculty, check for a non-null value on the rank column in the users view.

File: app/Services/UserRoleServices.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class UserRoleServices implements UserRoleContracts
+{
+    public function getRole(array $data): array
+    {
+        $userRole = UserRoles::where('user_email', $data['user_email'])->first();
+        return [
+            'user_role' => $userRole->user_role,
+            'user_email' => $data['user_email']
```
Comment (mbendandour) [inline app/Services/UserRoleServices.php]: add display_name, remember_token, password, to the migration

**Review by liadGolan — DISMISSED** [review]
There are dd()s still in here and the service needs a test

File: app/Services/UserRoleServices.php (hunk 1 above)
Comment (frlzjosh) [reply app/Services/UserRoleServices.php]: this is resolved!

**Review by mbendandour — APPROVED** [review]
Added get_role/nr_cyndee.bellamy@csun.edu to my route and received the appropriate return of:
{
user_role: "comparison",
user_email: "nr_cyndee.bellamy@csun.edu",
display_name: "Cyndee Bellamy",
password: "yungbasedgod",
remember_token: "12394iresjntgruirf9eiokwj"
}

Unit tests passed

**Review by matthewfritz — CHANGES_REQUESTED** [review]
We should not be storing passwords or even creating tables that have `password` columns.

File: database/factories/UserRoleFactory.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+use Faker\Generator as Faker;
+
+/*
```
Comment (liadGolan) [inline database/factories/UserRoleFactory.php]: This comment block is unnecessary

File: phpunit.xml [hunk 3]
```diff
@@ -10,7 +10,10 @@
         stopOnFailure="false">
     <testsuites>
         <testsuite name="Application Test Suite">
-            <directory suffix="Test.php">./tests</directory>
+            <directory suffix="Test.php">./tests/phpunit</directory>
+        </testsuite>
+        <testsuite name="Controller Test">
+            <directory suffix="Test.php">./tests/phpunit/ControllerTests</directory>
         </testsuite>
```
Comment (liadGolan) [inline phpunit.xml:14]: There should be a testSuite specifically for service tests if there is one for controllers

File: phpunit.xml [hunk 4]
```diff
@@ -10,7 +10,10 @@
         stopOnFailure="false">
     <testsuites>
         <testsuite name="Application Test Suite">
```
Comment (liadGolan) [inline phpunit.xml]: If we are calling each testSuite individually, there should not be a general one that does them both

File: tests/phpunit/ServicesTests/UserRoleServiceTest.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\ServicesTests;
```
Comment (liadGolan) [inline tests/phpunit/ServicesTests/UserRoleServiceTest.php]: The other suite was marked as `ControllerTests` where as this one is `ServicesTests`

They should both be singular (i.e. `ServiceTests`)

**Review by liadGolan — CHANGES_REQUESTED** [review]
The logic is good, but there are some naming and other code conventions that should be consistent

File: database/factories/UserRoleFactory.php (hunk 2 above)
Comment (frlzjosh) [reply database/factories/UserRoleFactory.php]: 👍

File: phpunit.xml (hunk 3 above)
Comment (frlzjosh) [reply phpunit.xml:14]: 👍

File: phpunit.xml (hunk 4 above)
Comment (frlzjosh) [reply phpunit.xml]: u right!

File: tests/phpunit/ServicesTests/UserRoleServiceTest.php (hunk 5 above)
Comment (frlzjosh) [reply tests/phpunit/ServicesTests/UserRoleServiceTest.php]: gotcha

**Review by liadGolan — DISMISSED** [review]
When I ran phpunit, only one test was ran. There are two tests here

**Review by liadGolan — APPROVED** [review]
👍 All Good

**Comment by frlzjosh** [comment]
> We should not be storing passwords or even creating tables that have `password` columns.

I resolved it!

**Review by frlzjosh — COMMENTED** [review]
i made the changes!

**Review by matthewfritz — APPROVED** [review]

**Pull request merged**